OMEdit/OMEditGUI/Plotting/VariablesWidget.cpp

```diff
@@ -1237,13 +1282,33 @@ void VariablesWidget::plotVariables(const QModelIndex &index, qreal curveThickne
     if (!pPlotWindow) {
       pPlotWindow = MainWindow::instance()->getPlotWindowContainer()->getCurrentWindow();
     }
-    // if pPlotWindow is 0 then create a new plot window.
-    if (!pPlotWindow) {
+    // if the variable is not an array and
+    // pPlotWindow is 0 or the plot's type is PLOTARRAY or PLOTARRAYPARAMETRIC
+    // then create a new plot window.
+    if (!pVariablesTreeItem->isMainArray() &&
+            (!pPlotWindow || pPlotWindow->getPlotType()==PlotWindow::PLOTARRAY
+             || pPlotWindow->getPlotType()== PlotWindow::PLOTARRAYPARAMETRIC))
+    {
       bool state = MainWindow::instance()->getPlotWindowContainer()->blockSignals(true);
       MainWindow::instance()->getPlotWindowContainer()->addPlotWindow();
       MainWindow::instance()->getPlotWindowContainer()->blockSignals(state);
       pPlotWindow = MainWindow::instance()->getPlotWindowContainer()->getCurrentWindow();
     }
```

Comment:
In the above block we block the signals that is why `subWindowActivated` signal is not raised. You can remove the block with the following code,
```
MainWindow::instance()->getPlotWindowContainer()->addPlotWindow();
bool state = MainWindow::instance()->getPlotWindowContainer()->blockSignals(true);
mpVariablesTreeModel->setData(index, Qt::Checked, Qt::CheckStateRole);
MainWindow::instance()->getPlotWindowContainer()->blockSignals(state);
pPlotWindow = MainWindow::instance()->getPlotWindowContainer()->getCurrentWindow();
```
The same goes for the plot parametric block below.